plugins/woocommerce/src/Internal/DataStores/Orders/CustomOrdersTableController.php

```diff
@@ -606,5 +428,101 @@ private function register_post_type_for_order_placeholders(): void {
+		}
+	}
 
+	/**
+	 * Returns the HPOS setting for rendering HPOS vs Post setting block in Features section of the settings page.
+	 *
+	 * @param array $sync_status Details of sync status, includes pending count, and count when sync started.
+	 *
+	 * @return array Feature setting object.
+	 */
+	private function get_hpos_setting_for_feature( $sync_status ) {
+		$hpos_enabled            = $this->custom_orders_table_usage_is_enabled();
+		$plugin_info             = $this->features_controller->get_compatible_plugins_for_feature( 'custom_order_tables', true );
+		$plugin_incompat_warning = $this->plugin_util->generate_incompatible_plugin_feature_warning( 'custom_order_tables', $plugin_info );
+		$sync_complete           = 0 === $sync_status['current_pending_count'];
+		$disabled_option         = array();
+		if ( count( array_merge( $plugin_info['uncertain'], $plugin_info['incompatible'] ) ) > 0 ) {
+			$disabled_option = array( 'yes' );
+		}
+		if ( ! $sync_complete ) {
+			$disabled_option = array( 'yes', 'no' );
+		}
+
+		return array(
+			'id'          => self::CUSTOM_ORDERS_TABLE_USAGE_ENABLED_OPTION,
+			'title'       => __( 'Data storage for orders', 'woocommerce' ),
+			'type'        => 'radio',
+			'options'     => array(
+				'no'  => __( 'WordPress Post tables', 'woocommerce' ),
+				'yes' => __( 'High performance order storage (new)', 'woocommerce' ),
+			),
+			'value'       => $hpos_enabled ? 'yes' : 'no',
+			'disabled'    => $disabled_option,
+			'desc'        => $plugin_incompat_warning,
+			'desc_at_end' => true,
+		);
+	}
+
+	/**
+	 * Returns the setting for rendering sync enabling setting block in Features section of the settings page.
+	 *
+	 * @param array $sync_status Details of sync status, includes pending count, and count when sync started.
+	 *
+	 * @return array Feature setting object.
+	 */
+	private function get_hpos_setting_for_sync( $sync_status ) {
+		$sync_in_progress = $this->batch_processing_controller->is_enqueued( get_class( $this->data_synchronizer ) );
+		$sync_enabled     = get_option( DataSynchronizer::ORDERS_DATA_SYNC_ENABLED_OPTION );
+		$sync_message     = '';
+		if ( $sync_in_progress && $sync_status['current_pending_count'] > 0 ) {
+			$sync_message = sprintf(
+				// translators: %d: number of pending orders.
+				__( 'Currently syncing orders... %d pending', 'woocommerce' ),
+				$sync_status['current_pending_count']
+			);
+		} elseif ( $sync_status['current_pending_count'] > 0 ) {
+			$sync_message = sprintf(
+				// translators: %d: number of pending orders.
+				__( 'Sync %d pending orders. You can switch data storage for orders only when posts and orders table are in sync.', 'woocommerce' ),
+				$sync_status['current_pending_count']
+			);
+		}
+
+		return array(
+			'id'       => DataSynchronizer::ORDERS_DATA_SYNC_ENABLED_OPTION,
+			'title'    => '',
+			'type'     => 'checkbox',
+			'desc'     => __( 'Keep posts and orders table in sync. (Compatibility mode)', 'woocommerce' ),
```

Comment:
Another nit, sorry 😬 ... Grammarly seems to prefer:

```suggestion
			'desc'     => __( 'Keep the posts and orders tables in sync (compatibility mode).', 'woocommerce' ),
```